### PR TITLE
read connection string from environment variable

### PR DESCRIPTION
**Pull request by pranshi06**
### What

The PR add the ability to use an Env Var to read the MSSQL connection string, instead of hard-coding the value in the config file (now the config file can contain the name of the env var, which can be provided while starting the server)

By default, the CLI will add the env var `CONNECTION_URI` for the string.

### CODE REVIEW

File: .github/workflows/cargo-test.yaml [hunk 1]
```diff
@@ -40,7 +40,7 @@ jobs:
         run: |
           cargo test --release --workspace
         env:
-          POSTGRESQL_CONNECTION_STRING: sqlserverql://sqlserver:password@localhost:64002
+          POSTGRESQL_CONNECTION_STRING: Server=localhost,64003;Uid=SA;Database=Chinook;Pwd=Password!;TrustServerCertificate=true
```
Comment (codingkarthik) [inline .github/workflows/cargo-test.yaml:43]: Update the env var name.

Comment (pranshi06) [reply .github/workflows/cargo-test.yaml]: My bad. Updated.

File: crates/configuration/src/version1.rs [hunk 2]
```diff
@@ -135,8 +158,21 @@ async fn select_first_row(
...
+    let connection_string = match &configuration.mssql_connection_string {
+        uri::ConnectionUri(Secret::Plain(s)) => s.to_owned(),
+        uri::ConnectionUri(Secret::FromEnvironment { variable }) => environment
+            .read(variable)
+            .map_err(|err| Error::MissingEnvironmentVariable {
+                file_path: file_path.to_path_buf(),
+                message: format!("{err}"),
+            })?,
+    };
+    let mssql_pool = create_mssql_pool(&connection_string).await.unwrap();
```
Comment (codingkarthik) [inline crates/configuration/src/version1.rs:175]: Can we avoid the use of `unwrap` here and instead throw an error?

**Review by codingkarthik — CHANGES_REQUESTED** [review]
Left a minor comment to review the unwrap

File: crates/configuration/src/version1.rs (hunk 2 above)
Comment (pranshi06) [reply crates/configuration/src/version1.rs]: Updated

**Pull request merged**